sys/net/gnrc/network_layer/icmpv6/echo/gnrc_icmpv6_echo.c

```diff
@@ -150,6 +150,12 @@ int gnrc_icmpv6_echo_send(const gnrc_netif_t *netif, const ipv6_addr_t *addr,
     ipv6_hdr_t *ipv6;
     uint8_t *databuf;
 
+    /* max IPv6 payload 65535 minus 8 bytes of icmp header = 65527 */
+    if (len > (int)(UINT16_MAX - sizeof(icmpv6_hdr_t))) {
```

Comment:
Casting to int here, might mask this check on 16-bit systems. `len` is unsigned `size`, so no cast should be necessary here.

Reply:
Agree. This cast is unnecessary.